e2e/src/recursive_search.spec.ts

```diff
@@ -1,12 +1,40 @@
+import { Diagnostic, DiagnosticSeverity, Position, Range } from 'vscode';
 import { assertDiagnostics } from './asserts';
-import { activateAndWait, getDocUri } from './helper';
+import { activateAndWait, getDocUri, insertText, replaceText } from './helper';
 
 suite('Recursive search', () => {
-  const DOC_URI = getDocUri('recursive_search.sql');
+  const RECURSIVE_SEARCH_DOC_URI = getDocUri('recursive_search.sql');
+  const TABLE_DOES_NOT_EXIST_DOC_URI = getDocUri('table_does_not_exist.sql');
 
   test('Should search model recursively if it is not found in BigQuery', async () => {
-    await activateAndWait(DOC_URI);
+    await activateAndWait(RECURSIVE_SEARCH_DOC_URI);
 
-    await assertDiagnostics(DOC_URI, []);
+    await assertDiagnostics(RECURSIVE_SEARCH_DOC_URI, []);
+  });
+
+  test('Should register created columns', async () => {
+    await activateAndWait(TABLE_DOES_NOT_EXIST_DOC_URI);
+    await insertText(new Position(6, 14), ', 2 as amount');
+
+    await activateAndWait(RECURSIVE_SEARCH_DOC_URI);
+    await replaceText('*', 's.amount');
+
+    await assertDiagnostics(RECURSIVE_SEARCH_DOC_URI, []);
+  });
+
+  test('Should unregister deleted columns', async () => {
```

Comment:
I don't like this test depends on previous. I now we have some similar cases, but we can start avoid such tests. Maybe you can prepare separate model to have independent test?